src/io/flutter/sdk/FlutterSdkUtil.java

```diff
@@ -35,13 +36,25 @@
 
 
 public class FlutterSdkUtil {
+  /** The environment variable to use to tell the flutter tool which app is driving it. */
+  public static final String FLUTTER_ENVIRONMENT = "FLUTTER_ENVIRONMENT";
```

Comment:
FLUTTER_CLIENT sounds good. Do you want to pass in the version separately? Could be FLUTTER_CLIENT_ID and FLUTTER_CLIENT_VERSION.

(Or perhaps FLUTTER_CALLER?)